### PR TITLE
Setup Sonar with badges

### LINKED ISSUE #33: Setup Sonar integration
[Sonar](https://www.sonarqube.org/) is an excellent 3rd party code quality tool. We should integrate it with the Github Actions to provide additional insight into code quality.

### PR DESCRIPTION
**Pull request by javajawa**
This is partial work on #33. Full Sonar setup will follow #11 to include coverage and quality gates.

### CODE REVIEW

**Comment by ajCameron** [comment]
https://sonarcloud.io/summary/overall?id=mewler_mewbot

Shows 26 code smells - but the sonarcloud bot is showing 0 -  not sure if this is a _them_ problem or an _us_ problem.

Looks good on the summary tho - so I think analysis is occuring successfully.

**Comment by javajawa** [comment]
For PRs, it only shows issues introduced in the changes; so if something gets approved as "good enough for now", it doesn't complain on every following change.
(It also supports assigning things and leaving comments in Sonar -- freely it's meant to be used as a complete tool chain in and off itself, but mostly I'm content to leave it complaining about security problems)

Kitsune raised a good point elsewhere of why the current listing isn't picking up those 26 issues, and the answer is it is, there's just all suppressed in the TOML file

File: .sonarcloud.properties [hunk 1]
```diff
@@ -0,0 +1,3 @@
+sonar.python.version = 3.8, 3.9, 3.10, 3.11
```
Comment (exachixkitsune) [inline .sonarcloud.properties:1]: Why 3.8 in here?

**Review by exachixkitsune — APPROVED** [review]

File: .sonarcloud.properties (hunk 1 above)
Comment (javajawa) [reply .sonarcloud.properties:1]: Matches https://github.com/mewler/mewbot/blob/mainline/setup.py#L30
I think 3.8 is what is specified everywhere as the requirement, although the linters do currently run (only) on 3.9

**Comment by ajCameron** [comment]
Cool! Then all good by me.

**Review by ajCameron — COMMENTED** [review]
Seems a sensible addition and the bot not reporting can be fixed later.

**Pull request merged**